docs/infrastructure/microk8s-quickstart.md

```diff
@@ -0,0 +1,136 @@
+
+## 3. Installing microk8s
+
+1. Make sure the machine is up to date and also has the `nfs-common` package installed
+
+   ```bash
+   sudo apt update && apt install nfs-common -y
+
+2. Install microk8s. Codacy is supported up to kubernetes 1.15. Therefore we recommend following the [official documentation on how to install microk8s](https://microk8s.io/docs/) with the `1.15/stable` channel.
+
+3. Allow privileged containers for microk8s
+
+   ```bash
+   sudo echo "--allow-privileged=true" >> /var/snap/microk8s/current/args/kube-apiserver
+   ```
+
+## 4. Configuring microk8s
+
+1. First, we must enable the following plugins on microk8s:
+
+    ```bash
+   microk8s.enable dns &&
+   microk8s.enable storage &&
+   microk8s.enable ingress &&
+   microk8s.status --wait-ready
+    ```
+
+2. Now that these plugins have been enabled, we should restart microk8s to ensure a proper bootstrap of the cluster
+
+    ```bash
+   microk8s.stop && microk8s.start && microk8s.status --wait-ready
+    ```
+
+    Further information on this subject can be [found here](https://github.com/ubuntu/microk8s/issues/493#issuecomment-498167435).
+
+3. The plugins are now enabled and the cluster bootstrapped. However, we must still wait for some microk8s internals (dns, http, and ingress) plugins to be ready. Failing to do so can result in pods entering a `CrashLoopBackoff` state:
+
+   ```bash
+    microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=kube-dns
+
+    microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=hostpath-provisioner
+
+    microk8s.kubectl wait -n ingress --for=condition=Ready pod -l name=nginx-ingress-microk8s
+    ```
+
+After these commands return successfully, we have ensured that dns, http, and nginx ingress are enabled and working properly inside the cluster.
+
+## 5. Installing Codacy
+
+### 1. Helm and Tiller
+
+First we must install `helm` onto the cluster. With `helm`, we can easily install charts and manage the lifecycle of the installed artifacts. One example is the ability to rollback a failed install.
+
+1. __Codacy supports up to `helm` version 2.16.3.__
+   __We currently do not support `helm` v3.__
+2. Follow the instructions on installing the `helm` client on the [Preparing to install Codacy documentation](../../docs/index.md).
+
+3. We must now deploy `tiller` onto the cluster.
+   1. First we must set a `serviceaccount` and a `clusterrolebinding` for `tiller` on the microk8s cluster:
+
+      ```bash
+      microk8s.kubectl create serviceaccount --namespace kube-system tiller
+      microk8s.kubectl create clusterrolebinding tiller-cluster-rule --clusterrole=cluster-admin --serviceaccount=kube-system:tiller
+      ```
+
+   2. Then we can install `tiller`, which will be installed with the same `HELM_VERSION` that you have specified earlier and we wait for tiller to be ready:
+
+      ```bash
+      helm init --service-account tiller
```

Comment:
I was having some troubles with this and had to do
`mkdir ~/.kube && sudo chown -Rv ubuntu:ubuntu ~/.kube && sudo microk8s.config > .kube/config`
before
`helm init --service-account tiller`
to make it work